test/e2e/40_tests.bats

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bats
+
+function setup() {
+  # Load libraries in setup() to access BATS_* variables
+  load lib/env
+  load lib/helm
+  load lib/install
+  load lib/poll
+
+  kubectl create namespace "$E2E_NAMESPACE"
+  install_gitsrv
+  install_tiller
+  install_helm_operator_with_helm
+  kubectl create namespace "$DEMO_NAMESPACE"
+}
+
+@test "When test.enable is set, tests are run" {
+  # Apply the HelmRelease
+  kubectl apply -f "$FIXTURES_DIR/releases/test/success.yaml" >&3
+
+  # Wait for it to be released
+  poll_until_equals 'release deploy' 'True' "kubectl -n $DEMO_NAMESPACE get helmrelease/podinfo-helm-repository -o jsonpath='{.status.conditions[?(@.type==\"Released\")].status}'"
+
+  if [ $HELM_VERSION = "v2" ]; then
+    test_exists_jq_filter='.info.status | has("last_test_suite_run")'
```

Comment:
there is currently a race condition where the observed generation is set, immediately followed by the lock opening up for the next chart sync interval loop, and the informer cache is not updated yet with the observed generation change, which results in an upgrade, which causes "last_test_suite_run" to be unset, which causes this test to fail. This is resolved by #447 which runs a harmless dry run compare, instead of a direct upgrade.

Reply:
Fixed the race condition by using polling in the test.